File: Data Structures & Algorithms/climbing-stairs/submission-1.java
class Solution {
    public int climbStairs(int n) {
        return climbStairs(n, 0);
    }

    public int climbStairs(int n, int i) {
        if (i > n) return 0;
        if (i == n) return 1;
        else return climbStairs(n, i + 1) + climbStairs(n, i + 2);
    }


}

File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
int search(vector<int> &nums, int target) {
  int l{0};
  int r{static_cast<int>(nums.size()) - 1};

  while (l <= r) {
    int m = (r + l) / 2;

    if (nums[static_cast<size_t>(m)] < target) {
      l = m + 1;
    } else if (nums[static_cast<size_t>(m)] > target) {
      r = m - 1;
    } else {
      return m;
    }
  }
  return -1;
}

};
